code/game/dna/genes/vg_powers.dm

```diff
@@ -268,23 +286,41 @@
 	var/list/targets = new /list()
 	var/list/validtargets = new /list()
 	var/turf/T = get_turf(user)
+	var/list/mobs_in_view = user.get_visible_mobs()
```

Comment:
since you only use this list to check for the presence of items in, would it make more sense to just replace this with the applicable `view` call

Reply:
I'm expecting the `in` operator to do a check that costs `O(m + n)` time. Where m is the amount of mobs in sight and n the rest objects. In the current situation it will only cost `O(m + n) + m*O(M)` time. Instead of `m * O(m+n)` time